crates/iota-storage/src/indexes.rs

```diff
@@ -1367,6 +1367,15 @@ impl IndexStore {
                 })
             })
             .await
+            .map(|balances_map| {
+                Arc::new(
+                    (*balances_map)
+                        .clone()
+                        .into_iter()
+                        .filter(|(_, TotalBalance { num_coins, .. })| *num_coins > 0)
+                        .collect::<HashMap<_, _>>(),
+                )
```

Comment:
Nit: It would save the reallocation, and reconstruction of the map
```suggestion
            .map(|mut balances_map| {
                Arc::make_mut(&mut balances_map)
                    .retain(|_, TotalBalance { num_coins, .. }| *num_coins > 0);
                balances_map
```

Reply:
Didn't know about the `make_mut` function. Applied the change. Thanks.